src/trajectory/Interpolated.cpp

```diff
@@ -100,6 +100,17 @@ void Interpolated::evaluateDerivative(double _t, int _derivative,
     _tangentVector.setZero();
   }
 
+  if(_t==mWaypoints[0].t)
```

Comment:
I am not 100% sure that I follow. For (1), the input `Interpolated` is piecewise linear so we do not need any velocities [as you noted]. If that is consistent with your understanding, then I agree that we should revert those changes.